### ISSUE TITLE
unclear docker-compose upgrade procedure and errors after upgrade

### ISSUE THREAD

**Issue by rfx77** [sos-berlin/joc-cockpit#18]
Today i did an upgrade of my docker-compose project from 2.5.3 to 2.5.4.

the compose project ist like in the documentation.
i did:
`docker-compose down`
change .env
`docker-compose pull`
`docker-compose up -d`

after that joc did not start. it said that some colums are missing. 
i did:

`docker-compose exec joc /bin/sh -c /opt/sos-berlin.com/js7/joc/install/joc_install_tables.sh`

this fixed the missing columns and joc is running but when i restart joc i got this error:

`java.sql.SQLException: Duplicate entry '6b1ba1a4d7b2' for key 'UNIQUE_IOS_HOST'`

this is the data in the table:

```
MariaDB [js7db]> select * from INV_OPERATING_SYSTEMS;
+----+--------------+-------+--------------+-------------------+---------------------+
| ID | HOSTNAME     | NAME  | ARCHITECTURE | DISTRIBUTION      | MODIFIED            |
+----+--------------+-------+--------------+-------------------+---------------------+
|  1 | 9b171e666910 | Linux | amd64        | 5.15.0-41-generic | 2023-07-13 06:20:57 |
|  2 | ad3d54c92194 | Linux | amd64        | 5.15.0-41-generic | 2023-07-13 06:24:01 |
|  3 | a7b79da77df0 | Linux | amd64        | 5.15.0-41-generic | 2023-07-28 07:31:53 |
|  4 | 36c1c53e238c | Linux | amd64        | 5.15.0-41-generic | 2023-07-28 07:32:16 |
|  6 | b96dd00385ba | Linux | amd64        | 5.15.0-41-generic | 2023-07-28 15:12:52 |
|  8 | 4f29eb464ca1 | Linux | amd64        | 5.15.0-41-generic | 2023-07-28 15:13:17 |
|  9 | f80d40461c97 | Linux | amd64        | 5.15.0-41-generic | 2023-07-28 15:17:12 |
| 10 | 6b1ba1a4d7b2 | Linux | amd64        | 5.15.0-41-generic | 2023-07-28 15:18:11 |
+----+--------------+-------+--------------+-------------------+---------------------+


```

the key is in the table but why das joc generate the same key twice when it starts.

Maybe you could clarify how an minor version update with docker-compose is done right. 

From my interpretation of the current documentation a "down, version change, pull, up" cycle should be the way to go.

Greetings,
Franz

**Comment by sosap** [comment]
It seems that you are using dynamic hostnames for your containers, maybe not a good idea. You will receive new entries in this table on each container start. You can remove such entries in the Dashboard from the JOC Cockpit icon indicating an unused instance by selecting the action menu item "Remove from inventory".

Anyway, this shouldn't raise an SQL error. Please forward the JOC Cockpit's joc.log file that includes the error message.

**Comment by rfx77** [comment]
joc.log:

`2023-07-28T15:18:10,249 INFO  qtp363988129-88      c.s.a.c.SOSServicePermissionIam              - Login with Identity Service JOC-INITIAL successful.
2023-07-28T15:18:10,747 INFO  qtp363988129-91      c.s.j.c.e.EventService                       - Start EventBus
2023-07-28T15:18:11,258 WARN  qtp363988129-86      o.h.e.j.s.SqlExceptionHelper                 - SQL Error: 1062, SQLState: 23000
2023-07-28T15:18:11,260 ERROR qtp363988129-86      o.h.e.j.s.SqlExceptionHelper                 - (conn=3) Duplicate entry '6b1ba1a4d7b2' for key 'UNIQUE_IOS_HOST'
2023-07-28T15:18:11,264 INFO  qtp363988129-86      c.s.j.c.JOCDefaultResponse                   -
REQUEST: ./controllers
PARAMS: {"controllerId":""}
USER: root
2023-07-28T15:18:11,265 ERROR qtp363988129-86      c.s.j.c.JOCDefaultResponse                   - DBInvalidDataException: com.sos.joc.exceptions.DBInvalidDataException: could not execute statement: 1062 (conn=3) Duplicate entry '6b1ba1a4d7b2' for key 'UNIQUE_IOS_HOST'
com.sos.joc.exceptions.DBInvalidDataException: could not execute statement: 1062 (conn=3) Duplicate entry '6b1ba1a4d7b2' for key 'UNIQUE_IOS_HOST'
        at com.sos.joc.db.inventory.os.InventoryOperatingSystemsDBLayer.saveOrUpdateOSItem(InventoryOperatingSystemsDBLayer.java:96) ~[sos-webservices-joc-global-2.5.4.jar:?]
        at com.sos.joc.controllers.impl.ControllersResourceImpl.getControllerAnswers(ControllersResourceImpl.java:189) ~[sos-webservices-joc-2.5.4.jar:?]
        at com.sos.joc.controllers.impl.ControllersResourceImpl.getControllers(ControllersResourceImpl.java:143) ~[sos-webservices-joc-2.5.4.jar:?]
        at com.sos.joc.controllers.impl.ControllersResourceImpl.postJobschedulerInstances(ControllersResourceImpl.java:93) ~[sos-webservices-joc-2.5.4.jar:?]
        at com.sos.joc.controllers.impl.ControllersResourceImpl.postJobschedulerInstances(ControllersResourceImpl.java:50) ~[sos-webservices-joc-2.5.4.jar:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
        at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory.lambda$static$0(ResourceMethodInvocationHandlerFactory.java:52) ~[jersey-server-3.0.8.jar:?]
        at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:134) ~[jersey-server-3.0.8.jar:?]
        at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:177) ~[jersey-server-3.0.8.jar:?]
        at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$ResponseOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:176) ~[jersey-server-3.0.8.jar:?]
        at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:81) ~[jersey-server-3.0.8.jar:?]
        at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:478) ~[jersey-server-3.0.8.jar:?]
        at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:400) ~[jersey-server-3.0.8.jar:?]
        at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:81) ~[jersey-server-3.0.8.jar:?]
        at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:255) ~[jersey-server-3.0.8.jar:?]
        at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248) ~[jersey-common-3.0.8.jar:?]
        at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244) ~[jersey-common-3.0.8.jar:?]
        at org.glassfish.jersey.internal.Errors.process(Errors.java:292) ~[jersey-common-3.0.8.jar:?]
        at org.glassfish.jersey.internal.Errors.process(Errors.java:274) ~[jersey-common-3.0.8.jar:?]
        at org.glassfish.jersey.internal.Errors.process(Errors.java:244) ~[jersey-common-3.0.8.jar:?]
        at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:265) ~[jersey-common-3.0.8.jar:?]
        at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:234) ~[jersey-server-3.0.8.jar:?]
        at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:684) ~[jersey-server-3.0.8.jar:?]
        at org.glassfish.jersey.servlet.WebComponent.serviceImpl(WebComponent.java:394) ~[jersey-container-servlet-core-3.0.8.jar:?]
        at org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:346) ~[jersey-container-servlet-core-3.0.8.jar:?]
        at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:358) ~[jersey-container-servlet-core-3.0.8.jar:?]
        at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:311) ~[jersey-container-servlet-core-3.0.8.jar:?]
        at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:205) ~[jersey-container-servlet-core-3.0.8.jar:?]
        at org.eclipse.jetty.servlet.ServletHolder$NotAsync.service(ServletHolder.java:1410) ~[jetty-servlet-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:764) ~[jetty-servlet-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:529) ~[jetty-servlet-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:131) ~[jetty-server-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:578) ~[jetty-security-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:122) ~[jetty-server-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:223) ~[jetty-server-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1570) ~[jetty-server-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:221) ~[jetty-server-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1381) ~[jetty-server-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:176) ~[jetty-server-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:484) ~[jetty-servlet-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1543) ~[jetty-server-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:174) ~[jetty-server-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1303) ~[jetty-server-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:129) ~[jetty-server-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:192) ~[jetty-server-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.server.handler.HandlerList.handle(HandlerList.java:51) ~[jetty-server-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:122) ~[jetty-server-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.rewrite.handler.RewriteHandler.handle(RewriteHandler.java:301) ~[jetty-rewrite-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:122) ~[jetty-server-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.server.Server.handle(Server.java:563) ~[jetty-server-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.server.HttpChannel.lambda$handle$0(HttpChannel.java:505) ~[jetty-server-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:762) ~[jetty-server-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:497) ~[jetty-server-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:282) ~[jetty-server-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:314) ~[jetty-io-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:100) ~[jetty-io-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.io.SelectableChannelEndPoint$1.run(SelectableChannelEndPoint.java:53) ~[jetty-io-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.runTask(AdaptiveExecutionStrategy.java:416) ~[jetty-util-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.consumeTask(AdaptiveExecutionStrategy.java:385) ~[jetty-util-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.tryProduce(AdaptiveExecutionStrategy.java:272) ~[jetty-util-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.lambda$new$0(AdaptiveExecutionStrategy.java:140) ~[jetty-util-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:411) [jetty-util-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:969) [jetty-util-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.doRunJob(QueuedThreadPool.java:1194) [jetty-util-11.0.15.jar:11.0.15]
        at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1149) [jetty-util-11.0.15.jar:11.0.15]
        at java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: com.sos.commons.hibernate.exception.SOSHibernateObjectOperationException: could not execute statement: 1062 (conn=3) Duplicate entry '6b1ba1a4d7b2' for key 'UNIQUE_IOS_HOST'
        at com.sos.commons.hibernate.SOSHibernateSession.save(SOSHibernateSession.java:990) ~[sos-commons-hibernate-2.5.4.jar:?]
        at com.sos.joc.db.inventory.os.InventoryOperatingSystemsDBLayer.saveOrUpdateOSItem(InventoryOperatingSystemsDBLayer.java:80) ~[sos-webservices-joc-global-2.5.4.jar:?]
        ... 68 more
Caused by: org.hibernate.exception.ConstraintViolationException: could not execute statement
        at org.hibernate.exception.internal.SQLExceptionTypeDelegate.convert(SQLExceptionTypeDelegate.java:59) ~[hibernate-core-5.5.4.Final.jar:5.5.4.Final]
        at org.hibernate.exception.internal.StandardSQLExceptionConverter.convert(StandardSQLExceptionConverter.java:35) ~[hibernate-core-5.5.4.Final.jar:5.5.4.Final]
        at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:113) ~[hibernate-core-5.5.4.Final.jar:5.5.4.Final]
        at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:99) ~[hibernate-core-5.5.4.Final.jar:5.5.4.Final]
        at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:200) ~[hibernate-core-5.5.4.Final.jar:5.5.4.Final]
        at org.hibernate.id.insert.AbstractSelectingDelegate.performInsert(AbstractSelectingDelegate.java:45) ~[hibernate-core-5.5.4.Final.jar:5.5.4.Final]
        at org.hibernate.persister.entity.AbstractEntityPersister.insert(AbstractEntityPersister.java:3277) ~[hibernate-core-5.5.4.Final.jar:5.5.4.Final]
        at org.hibernate.persister.entity.AbstractEntityPersister.insert(AbstractEntityPersister.java:3883) ~[hibernate-core-5.5.4.Final.jar:5.5.4.Final]
        at org.hibernate.internal.StatelessSessionImpl.insert(StatelessSessionImpl.java:110) ~[hibernate-core-5.5.4.Final.jar:5.5.4.Final]
        at org.hibernate.internal.StatelessSessionImpl.insert(StatelessSessionImpl.java:89) ~[hibernate-core-5.5.4.Final.jar:5.5.4.Final]
        at com.sos.commons.hibernate.SOSHibernateSession.save(SOSHibernateSession.java:981) ~[sos-commons-hibernate-2.5.4.jar:?]
        at com.sos.joc.db.inventory.os.InventoryOperatingSystemsDBLayer.saveOrUpdateOSItem(InventoryOperatingSystemsDBLayer.java:80) ~[sos-webservices-joc-global-2.5.4.jar:?]
        ... 68 more
Caused by: java.sql.SQLIntegrityConstraintViolationException: (conn=3) Duplicate entry '6b1ba1a4d7b2' for key 'UNIQUE_IOS_HOST'
        at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.createException(ExceptionFactory.java:70) ~[mariadb-java-client-2.7.2.jar:?]
        at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.create(ExceptionFactory.java:153) ~[mariadb-java-client-2.7.2.jar:?]
        at org.mariadb.jdbc.MariaDbStatement.executeExceptionEpilogue(MariaDbStatement.java:274) ~[mariadb-java-client-2.7.2.jar:?]
        at org.mariadb.jdbc.ClientSidePreparedStatement.executeInternal(ClientSidePreparedStatement.java:229) ~[mariadb-java-client-2.7.2.jar:?]
        at org.mariadb.jdbc.ClientSidePreparedStatement.execute(ClientSidePreparedStatement.java:149) ~[mariadb-java-client-2.7.2.jar:?]
        at org.mariadb.jdbc.ClientSidePreparedStatement.executeUpdate(ClientSidePreparedStatement.java:181) ~[mariadb-java-client-2.7.2.jar:?]
        at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61) ~[HikariCP-3.2.0.jar:?]
        at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java) ~[HikariCP-3.2.0.jar:?]
        at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:197) ~[hibernate-core-5.5.4.Final.jar:5.5.4.Final]
        at org.hibernate.id.insert.AbstractSelectingDelegate.performInsert(AbstractSelectingDelegate.java:45) ~[hibernate-core-5.5.4.Final.jar:5.5.4.Final]
        at org.hibernate.persister.entity.AbstractEntityPersister.insert(AbstractEntityPersister.java:3277) ~[hibernate-core-5.5.4.Final.jar:5.5.4.Final]
        at org.hibernate.persister.entity.AbstractEntityPersister.insert(AbstractEntityPersister.java:3883) ~[hibernate-core-5.5.4.Final.jar:5.5.4.Final]
        at org.hibernate.internal.StatelessSessionImpl.insert(StatelessSessionImpl.java:110) ~[hibernate-core-5.5.4.Final.jar:5.5.4.Final]
        at org.hibernate.internal.StatelessSessionImpl.insert(StatelessSessionImpl.java:89) ~[hibernate-core-5.5.4.Final.jar:5.5.4.Final]
        at com.sos.commons.hibernate.SOSHibernateSession.save(SOSHibernateSession.java:981) ~[sos-commons-hibernate-2.5.4.jar:?]
        at com.sos.joc.db.inventory.os.InventoryOperatingSystemsDBLayer.saveOrUpdateOSItem(InventoryOperatingSystemsDBLayer.java:80) ~[sos-webservices-joc-global-2.5.4.jar:?]
        ... 68 more
Caused by: org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException: Duplicate entry '6b1ba1a4d7b2' for key 'UNIQUE_IOS_HOST'
        at org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException.of(MariaDbSqlException.java:34) ~[mariadb-java-client-2.7.2.jar:?]
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.exceptionWithQuery(AbstractQueryProtocol.java:192) ~[mariadb-java-client-2.7.2.jar:?]
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.exceptionWithQuery(AbstractQueryProtocol.java:175) ~[mariadb-java-client-2.7.2.jar:?]
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:319) ~[mariadb-java-client-2.7.2.jar:?]
        at org.mariadb.jdbc.ClientSidePreparedStatement.executeInternal(ClientSidePreparedStatement.java:220) ~[mariadb-java-client-2.7.2.jar:?]
        at org.mariadb.jdbc.ClientSidePreparedStatement.execute(ClientSidePreparedStatement.java:149) ~[mariadb-java-client-2.7.2.jar:?]
        at org.mariadb.jdbc.ClientSidePreparedStatement.executeUpdate(ClientSidePreparedStatement.java:181) ~[mariadb-java-client-2.7.2.jar:?]
        at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61) ~[HikariCP-3.2.0.jar:?]
        at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java) ~[HikariCP-3.2.0.jar:?]
        at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:197) ~[hibernate-core-5.5.4.Final.jar:5.5.4.Final]
        at org.hibernate.id.insert.AbstractSelectingDelegate.performInsert(AbstractSelectingDelegate.java:45) ~[hibernate-core-5.5.4.Final.jar:5.5.4.Final]
        at org.hibernate.persister.entity.AbstractEntityPersister.insert(AbstractEntityPersister.java:3277) ~[hibernate-core-5.5.4.Final.jar:5.5.4.Final]
        at org.hibernate.persister.entity.AbstractEntityPersister.insert(AbstractEntityPersister.java:3883) ~[hibernate-core-5.5.4.Final.jar:5.5.4.Final]
        at org.hibernate.internal.StatelessSessionImpl.insert(StatelessSessionImpl.java:110) ~[hibernate-core-5.5.4.Final.jar:5.5.4.Final]
        at org.hibernate.internal.StatelessSessionImpl.insert(StatelessSessionImpl.java:89) ~[hibernate-core-5.5.4.Final.jar:5.5.4.Final]
        at com.sos.commons.hibernate.SOSHibernateSession.save(SOSHibernateSession.java:981) ~[sos-commons-hibernate-2.5.4.jar:?]
        at com.sos.joc.db.inventory.os.InventoryOperatingSystemsDBLayer.saveOrUpdateOSItem(InventoryOperatingSystemsDBLayer.java:80) ~[sos-webservices-joc-global-2.5.4.jar:?]
        ... 68 more
Caused by: java.sql.SQLException: Duplicate entry '6b1ba1a4d7b2' for key 'UNIQUE_IOS_HOST'
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readErrorPacket(AbstractQueryProtocol.java:1681) ~[mariadb-java-client-2.7.2.jar:?]
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readPacket(AbstractQueryProtocol.java:1543) ~[mariadb-java-client-2.7.2.jar:?]
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.getResult(AbstractQueryProtocol.java:1506) ~[mariadb-java-client-2.7.2.jar:?]
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:316) ~[mariadb-java-client-2.7.2.jar:?]
        at org.mariadb.jdbc.ClientSidePreparedStatement.executeInternal(ClientSidePreparedStatement.java:220) ~[mariadb-java-client-2.7.2.jar:?]
        at org.mariadb.jdbc.ClientSidePreparedStatement.execute(ClientSidePreparedStatement.java:149) ~[mariadb-java-client-2.7.2.jar:?]
        at org.mariadb.jdbc.ClientSidePreparedStatement.executeUpdate(ClientSidePreparedStatement.java:181) ~[mariadb-java-client-2.7.2.jar:?]
        at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61) ~[HikariCP-3.2.0.jar:?]
        at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java) ~[HikariCP-3.2.0.jar:?]
        at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:197) ~[hibernate-core-5.5.4.Final.jar:5.5.4.Final]
        at org.hibernate.id.insert.AbstractSelectingDelegate.performInsert(AbstractSelectingDelegate.java:45) ~[hibernate-core-5.5.4.Final.jar:5.5.4.Final]
        at org.hibernate.persister.entity.AbstractEntityPersister.insert(AbstractEntityPersister.java:3277) ~[hibernate-core-5.5.4.Final.jar:5.5.4.Final]
        at org.hibernate.persister.entity.AbstractEntityPersister.insert(AbstractEntityPersister.java:3883) ~[hibernate-core-5.5.4.Final.jar:5.5.4.Final]
        at org.hibernate.internal.StatelessSessionImpl.insert(StatelessSessionImpl.java:110) ~[hibernate-core-5.5.4.Final.jar:5.5.4.Final]
        at org.hibernate.internal.StatelessSessionImpl.insert(StatelessSessionImpl.java:89) ~[hibernate-core-5.5.4.Final.jar:5.5.4.Final]
        at com.sos.commons.hibernate.SOSHibernateSession.save(SOSHibernateSession.java:981) ~[sos-commons-hibernate-2.5.4.jar:?]
        at com.sos.joc.db.inventory.os.InventoryOperatingSystemsDBLayer.saveOrUpdateOSItem(InventoryOperatingSystemsDBLayer.java:80) ~[sos-webservices-joc-global-2.5.4.jar:?]
        ... 68 more
2023-07-28T15:37:16,296 INFO  qtp363988129-103     c.s.a.c.SOSServicePermissionIam              - Client Certificate CN read from Login: n/a
2023-07-28T15:37:16,687 INFO  qtp363988129-103     c.s.a.c.SOSServicePermissionIam              - Login with Identity Service JOC-INITIAL successful.
2023-07-28T15:37:53,712 INFO  main                 c.s.j.s.JocServletContainer                  - cleanup log files: /var/sos-berlin.com/js7/joc/logs
2023-07-28T15:37:54,068 INFO  main                 c.s.j.Globals                                - Java version = 11.0.20
2023-07-28T15:37:54,092 INFO  main                 c.s.j.Globals                                - JOC Cockpit version = 2.5.4
2023-07-28T15:37:54,093 INFO  main                 c.s.j.Globals                                - API schema version = 2.5.4
2023-07-28T15:37:54,094 INFO  main                 c.s.j.Globals                                - Inventory schema version = 1.5.4
2023-07-28T15:37:54,095 INFO  main                 c.s.j.Globals                                - Security Level = LOW
2023-07-28T15:37:54,215 INFO  main                 o.h.Version                                  - HHH000412: Hibernate ORM core version 5.5.4.Final
2023-07-28T15:37:54,845 INFO  main                 o.h.a.c.Version                              - HCANN000001: Hibernate Commons Annotations {5.1.2.Final}
2023-07-28T15:37:55,037 INFO  main                 o.h.d.Dialect                                - HHH000400: Using dialect: org.hibernate.dialect.MySQLInnoDBDialect
2023-07-28T15:37:57,285 INFO  main                 o.h.e.j.c.i.ConnectionProviderInitiator      - HHH000130: Instantiating explicit connection provider: org.hibernate.hikaricp.internal.HikariCPConnectionProvider
2023-07-28T15:37:57,303 INFO  main                 c.z.h.HikariDataSource                       - HikariPool-1 - Starting...
2023-07-28T15:37:57,450 INFO  main                 c.z.h.HikariDataSource                       - HikariPool-1 - Start completed.
2023-07-28T15:37:58,406 INFO  main                 j.p.j.JProxyContext                          - JS7 Proxy 2.5.4
2023-07-28T15:37:58,769 INFO  main                 c.s.j.c.p.Proxies                            - starting all proxies for user JOC ...
2023-07-28T15:37:58,956 INFO  main                 c.s.j.c.p.ControllerApiContext               - connect ControllerApi of 'jobscheduler' (http://controller:4444)
2023-07-28T15:38:00,257 INFO  main                 c.s.j.c.p.ProxyContext                       - start Proxy of 'jobscheduler' (http://controller:4444)
2023-07-28T15:38:07,313 INFO  JControllerProxy-32  c.s.j.c.p.ProxyContext                       - 'jobscheduler' (http://controller:4444): ProxyCoupled(1690557426978003)
2023-07-28T15:38:14,116 INFO  main                 c.s.j.Globals                                - HTTP(S) connection timeout = 2s
2023-07-28T15:38:14,116 INFO  main                 c.s.j.Globals                                - HTTP(S) socket timeout = 5s
2023-07-28T15:38:14,117 INFO  main                 c.s.j.Globals                                - HTTPS with hostname verification in certificate = true
2023-07-28T15:38:14,126 INFO  main                 c.s.j.c.w.WorkflowPaths                      - ... init workflow name->path mapping
2023-07-28T15:38:14,214 INFO  main                 c.s.j.c.a.AgentStoreUtils                    - AgentStoreUtils has been registered for the EventBus.
2023-07-28T15:38:14,216 INFO  commonPool-worker-23 c.s.c.u.SOSShell                             - [SYSTEM] Linux, version=5.15.0-41-generic, arch=amd64
2023-07-28T15:38:14,224 INFO  commonPool-worker-23 c.s.c.u.SOSShell                             - [JVM][89@f80d40461c97][pid=89][version=11.0.20][vm=Alpine OpenJDK 64-Bit Server VM][available processors=12][memory max=256 MB,total=256 MB,free=197.97 MB][input arguments=[-Djava.io.tmpdir=/var/sos-berlin.com/js7/joc/temp, -Djetty.home=/opt/sos-berlin.com/js7/joc/jetty, -Djetty.base=/var/sos-berlin.com/js7/joc, -Xmx256m, -Djava.io.tmpdir=/var/sos-berlin.com/js7/joc/temp, -Dfile.encoding=UTF-8, -Duser.timezone=UTC, -Djava.util.prefs.userRoot=/var/sos-berlin.com/js7/joc]]
`

**Comment by rfx77** [comment]
Could you also please clarify the docker-compose update procedure
is it necessary to run the joc_install_tables.sh script?

**Comment by sosap** [comment]
Thank you, the log output confirms the bug. 

Yes, it is required to run the joc_install_table.sh script to update the database for use with newer releases, see [JS7 - JOC Cockpit Installation for Containers](https://kb.sos-berlin.com/display/JS7/JS7+-+JOC+Cockpit+Installation+for+Containers). The image is prepared for use with a number of DBMS, therefore users have to apply updates to the database in their DBMS from running the script once.

This is different if you build your own images, see [JS7 - JOC Cockpit Build of Container Image](https://kb.sos-berlin.com/display/JS7/JS7+-+JOC+Cockpit+Build+of+Container+Image).

**Comment by sosap** [comment]
The issue is confirmed and is tracked by

* [JOC-1586: Constraint violation when storing to the table INV_OPERATING_SYSTEMS](https://change.sos-berlin.com/browse/JOC-1586)

The fix will become available with releases 2.5.5 and 2.6.2.
